azavea/echo-locator
# Add management command to migrate housing seeker data

## Overview

This PR:
- updates the `Destination` model and adds a migration so that a `PointField` of `location` replaces `address`
- adds `boto3` as a backend dependency
- exposes `AWS_PROFILE` env var for django service in `docker-compose`, which should make local development easier if the developer has multiple aws profiles. This should not matter for staging and prod
- adds a management command to migrate housing seeker data from S3 to the postgres db:
```
required arguments:
  -b BUCKET, --bucket BUCKET
                        S3 bucket to read housing seeker profile from
  -p PREFIX, --prefix PREFIX
                        S3 prefix under the bucket that stores housing seeker profile
  -vkts VOUCHER_KEYS_TO_SKIP [VOUCHER_KEYS_TO_SKIP ...], --voucher_keys_to_skip VOUCHER_KEYS_TO_SKIP [VOUCHER_KEYS_TO_SKIP ...]
                        Space separated list of double-quoted voucher numbers.
```

### Notes

See attached original issue for discussions about assumptions and edge case handling.


## Testing Instructions

 * `./scripts/update`
 * `AWS_PROFILE=<your echo profile name>./scripts/manage migrate_housing_seeker_data -b <staging profile bucket name> -p public -vkts ""`
 * The above should work and should insert related staging profiles as records in your local DB
 * Make sure it has an output of summary
 * `AWS_PROFILE=<your echo profile name>./scripts/manage migrate_housing_seeker_data -b <production profile bucket name> -p public -vkts ""`
 * The above should work and should insert related production profiles as records in your local DB
 * Make sure it has an output of summary

Closes #432
